mmv1/third_party/terraform/sweeper/hashi_sweeper_fork.go

```diff
@@ -0,0 +1,274 @@
+package sweeper
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+// flagSweep is a flag available when running tests on the command line. It
+// contains a comma separated list of regions to for the sweeper functions to
+// run in.  This flag bypasses the normal Test path and instead runs functions designed to
+// clean up any leaked resources a testing environment could have created. It is
+// a best effort attempt, and relies on Provider authors to implement "Sweeper"
+// methods for resources.
+
+// Adding Sweeper methods with AddTestSweepers will
+// construct a list of sweeper funcs to be called here. We iterate through
+// regions provided by the sweep flag, and for each region we iterate through the
+// tests, and exit on any errors. At time of writing, sweepers are ran
+// sequentially, however they can list dependencies to be ran first. We track
+// the sweepers that have been ran, so as to not run a sweeper twice for a given
+// region.
+//
+// WARNING:
+// Sweepers are designed to be destructive. You should not use the -sweep flag
+// in any environment that is not strictly a test environment. Resources will be
+// destroyed.
+
+var (
+	flagSweep              *string
+	flagSweepAllowFailures *bool
+	flagSweepRun           *string
+	sweeperFuncs           map[string]*Sweeper
+)
+
+// SweeperFunc is a signature for a function that acts as a sweeper. It
+// accepts a string for the region that the sweeper is to be ran in. This
+// function must be able to construct a valid client for that region.
+type SweeperFunc func(r string) error
+
+type Sweeper struct {
+	// Name for sweeper. Must be unique to be ran by the Sweeper Runner
+	Name string
+
+	// Dependencies list the const names of other Sweeper functions that must be ran
+	// prior to running this Sweeper. This is an ordered list that will be invoked
+	// recursively at the helper/resource level
+	Dependencies []string
+
+	// Sweeper function that when invoked sweeps the Provider of specific
+	// resources
+	F SweeperFunc
+}
+
+func init() {
+	sweeperFuncs = make(map[string]*Sweeper)
+}
+
+// registerFlags checks for and gets existing flag definitions before trying to redefine them.
+// This is needed because this package and terraform-plugin-testing both define the same sweep flags.
+// By checking first, we ensure we reuse any existing flags rather than causing a panic from flag redefinition.
+// This allows this module to be used alongside terraform-plugin-testing without conflicts.
```

Comment:
This logic is outdated. I'm going to refactor this in a sequential PR after making sure this pattern works.